### PR TITLE
Feat/478 custom derivation settings

### PR DESCRIPTION
**Pull request by umurb**
Need to work on Derivation Path Validations.

A valid derivation path should start with;
`44'/1729'/`
and should be followed by 2 or 3 integers
`44'/1729'/X'/X'` or `44'/1729'/X'/X'/X'`

### CODE REVIEW

File: app/containers/HomeSettings/index.js [hunk 1]
```diff
@@ -213,6 +239,39 @@ class SettingsPage extends Component<Props> {
...
 
+  renderPaths(paths, selectedPath) {
+    const { theme } = this.props;
+    return paths.map((path, index) => {
+      const label = path.get('label');
+      const derivation = path.get('derivation');
+      const selected = selectedPath === label;
+      const option = (
+        <SelectOption>
+          <OptionStatus>
```
Comment (developer0623) [inline app/containers/HomeSettings/index.js]: Please use the `selected` variable on `OptionStatus`

File: app/containers/HomeSettings/index.js [hunk 2]
```diff
@@ -180,6 +194,18 @@ class SettingsPage extends Component<Props> {
...
+      const label = path.get('label');
+      return label === selectedPath;
+    });
+    if (foundPath) {
+      path = foundPath.get('derivation');
+    }
+    return path;
+  };
+
   renderNodes(nodes, selectedNode) {
```
Comment (developer0623) [inline app/containers/HomeSettings/index.js]: Please use `arrow` function

File: app/containers/HomeSettings/index.js [hunk 3]
```diff
@@ -213,6 +239,39 @@ class SettingsPage extends Component<Props> {
     });
   }
 
+  renderPaths(paths, selectedPath) {
```
Comment (developer0623) [inline app/containers/HomeSettings/index.js]: Please use `arrow` function

File: app/reduxContent/settings/reducers.js [hunk 4]
```diff
@@ -82,6 +91,46 @@ export default handleActions(
...
+      return state;
+    },
+    [REMOVE_PATH]: (state, action) => {
+      const { label } = action;
+      const pathsList = state.get('pathsList');
+      const indexFound = pathsList.findIndex(item => {
+        return item.get('label') === label;
+      });
+
+      if (indexFound >= -1) {
```
Comment (developer0623) [inline app/reduxContent/settings/reducers.js]: Please change `>=` to `>`

File: app/reduxContent/settings/reducers.js [hunk 5]
```diff
@@ -82,6 +91,46 @@ export default handleActions(
...
+      return state;
+    },
+    [UPDATE_PATH]: (state, action) => {
+      const newPath = action.path;
+      const pathsList = state.get('pathsList');
+      const indexFound = pathsList.findIndex(item => {
+        return item.get('label') === newPath.label;
+      });
+
+      if (indexFound >= -1) {
```
Comment (developer0623) [inline app/reduxContent/settings/reducers.js]: Please change `>=` to `>`

File: app/reduxContent/settings/reducers.js [hunk 6]
```diff
@@ -82,6 +91,46 @@ export default handleActions(
       }
       return state;
     },
+    [ADD_PATH]: (state, action) => {
+      const newPath = action.path;
+      const pathsList = state.get('pathsList');
+      const indexFound = pathsList.findIndex(item => {
```
Comment (developer0623) [inline app/reduxContent/settings/reducers.js]: const indexFound = pathsList.findIndex(item => {
        return item.get('label') === newPath.label;
      });

to
const indexFound = pathsList.findIndex(item => item.get('label') === newPath.label);

File: app/reduxContent/settings/reducers.js [hunk 7]
```diff
@@ -82,6 +91,46 @@ export default handleActions(
...
+
+      if (indexFound >= -1) {
+        return state.set('pathsList', pathsList.splice(indexFound, 1));
+      }
+      return state;
+    },
+    [UPDATE_PATH]: (state, action) => {
+      const newPath = action.path;
+      const pathsList = state.get('pathsList');
+      const indexFound = pathsList.findIndex(item => {
```
Comment (developer0623) [inline app/reduxContent/settings/reducers.js]: const indexFound = pathsList.findIndex(item => {
        return item.get('label') === newPath.label;
      });
to
const indexFound = pathsList.findIndex(item => item.get('label') === newPath.label);

File: app/containers/HomeSettings/index.js (hunk 1 above)
Comment (umurb) [reply app/containers/HomeSettings/index.js]: Should I do the same changes to

const selected = selectedNode === name;
      const option = (
        <SelectOption>
          <OptionStatus>
            {selected ? (
              <Check

Comment (developer0623) [reply app/containers/HomeSettings/index.js]: No, there is no need for <Check>

**Pull request merged**